message/observe.go

```diff
@@ -0,0 +1,33 @@
+// package message defines the terminology required for sharing
+// data between multiple PATH instances.
+package message
+
+// ObservationSet defines the functionality required for sharing
+// data between PATH instances.
+// e.g. QoS instances can share data by supplying an implementation
+// of this interface.
+type ObservationSet interface {
+	// MarshalJSON returns the serialized form
+	// of the set of observations, in JSON format.
+	// This is required for sharing QoS data between
+	// multiple PATH instances.
+	MarshalJSON() ([]byte, error)
```

Comment:
What are your thoughts on using `proto` instead of `JSON` for observations?

Reply:
`proto` may be a better choice here.
There is a comment in the PR on that, just updated it to `TODO_UPNEXT` to assess this option.
`JSON` was mainly used to allow faster implementation of the required support.